servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -1878,6 +1878,8 @@ void RenderForwardClustered::_render_scene(RenderDataRD *p_render_data, const Co
 		continue_depth = !finish_depth;
 	}
 
+	_process_render_effects(RS::RENDERING_EFFECT_CALLBACK_TYPE_PRE_OPAQUE, p_render_data);
```

Comment:
We are going to have to figure out some way to transition resource states. As you can see here, there is no guarantee that the user can access depth as the depth may have been continued or finished here

Reply:
Yeah and that is going to be a problem if we change things radically with some of the redesign Reduz is planning. With the acyclic graph approach you'd just be inserting things and those decisions are made later, but we'll seriously break compatibility.

Reply:
I added a few settings to the structure that should allow access to the depth buffer now. Though there is still room for improvement here. What makes this extra challenging is that if in rendering effects we will use a framebuffer to continue with the color and depth buffers, there isn't a really good way to steer that, especially with how this can already change depending on other effects being enabled.

I'm hoping this is something that can be done much more automatic when @reduz works out the acyclic graph implementation.

For now I'm hoping that what we're doing is "good enough" and we'll see what needs to be improved as we get more solid use cases for this logic.

Reply:
This isn't a concern anymore with the ARG